### PR TITLE
M9 pull request

### PR DESCRIPTION
**Pull request by y-tst**
Got rid: drivers .exe files
Added: model, listener (incl. screen capture), work with properties, test
Test suite - smokeTestForOverDriveLibrary.xml

### CODE REVIEW

File: src/test/java/Model/User.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package Model;
+
+public class User {
+
+    private String username;
+    private String password;
+
```
Comment (olelysieev-clgx) [inline src/test/java/Model/User.java]: extra empty line

File: src/test/java/pages/overDrivePages/OverDriveHomePage.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package pages.overDrivePages;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import pages.BasePage;
+
+public class OverDriveHomePage extends BasePage {
+
+    public static final String OVERDRIVE_HOMEPAGE_URL = "https://www.overdrive.com/";
```
Comment (olelysieev-clgx) [inline src/test/java/pages/overDrivePages/OverDriveHomePage.java]: private

File: src/test/java/pages/overDrivePages/OverDriveHomePage.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    public OverDriveHomePage pageOpener() {
+        driver.get(OVERDRIVE_HOMEPAGE_URL);
+        waitForPageLoad();
+
+        return this;
+    }
+
+    public String userNameOnOverDrivePage() {
+        waitForElementVisibility(userIcon);
+        return userIcon.getText();
```
Comment (olelysieev-clgx) [inline src/test/java/pages/overDrivePages/OverDriveHomePage.java:32]: add empty line before return statement

File: src/test/java/pages/overDrivePages/OverDriveSignInPage.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+import Model.User;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import pages.BasePage;
+
+public class OverDriveSignInPage extends BasePage {
+
+    public static final String OVERDRIVE_SIGN_IN_PAGE_URL = "https://www.overdrive.com/account/sign-in?returnUrl=%2Faccount%2Fsign-in";
```
Comment (olelysieev-clgx) [inline src/test/java/pages/overDrivePages/OverDriveSignInPage.java]: private

File: src/test/java/tests/BaseTest.java [hunk 5]
```diff
@@ -1,30 +1,38 @@
...
 import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.edge.EdgeDriver;
 import org.openqa.selenium.firefox.FirefoxDriver;
 import org.testng.annotations.*;
+import service.TestListener;
 
 import java.util.logging.Logger;
 
+@Listeners({TestListener.class})
+
```
Comment (olelysieev-clgx) [inline src/test/java/tests/BaseTest.java:14]: remove empty line between @ and class name

File: src/test/java/service/TestListener.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+import tests.BaseTest;
+
+import java.io.File;
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
```
Comment (olelysieev-clgx) [inline src/test/java/service/TestListener.java:17]: empty line

File: src/test/java/service/TestDataReader.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package service;
+
+import java.util.ResourceBundle;
+
+public class TestDataReader {
+    private static ResourceBundle resourceBundle = ResourceBundle.getBundle(System.getProperty("environment", "qa"));
```
Comment (olelysieev-clgx) [inline src/test/java/service/TestDataReader.java:6]: I suppose, everywhere data will be get just from qa env....
but we need also from 'dev' env....

File: src/test/java/Model/User.java (hunk 1 above)
Comment (y-tst) [reply src/test/java/Model/User.java]: fixed

File: src/test/java/pages/overDrivePages/OverDriveHomePage.java (hunk 2 above)
Comment (y-tst) [reply src/test/java/pages/overDrivePages/OverDriveHomePage.java]: done

File: src/test/java/pages/overDrivePages/OverDriveSignInPage.java (hunk 4 above)
Comment (y-tst) [reply src/test/java/pages/overDrivePages/OverDriveSignInPage.java]: +

File: src/test/java/tests/BaseTest.java (hunk 5 above)
Comment (y-tst) [reply src/test/java/tests/BaseTest.java:14]: done

File: src/test/java/service/TestDataReader.java (hunk 7 above)
Comment (y-tst) [reply src/test/java/service/TestDataReader.java:6]: both properties are used

File: src/test/java/service/TestListener.java (hunk 6 above)
Comment (y-tst) [reply src/test/java/service/TestListener.java:17]: yep

**Review by olelysieev-clgx — APPROVED** [review]

**Pull request merged**